src/session_impl.cpp

```diff
@@ -6605,9 +6605,18 @@ namespace aux {
 		set_external_address(ip, source_dht, source);
 	}
 
-	address session_impl::external_address()
+	address session_impl::external_address(address_type at)
 	{
-		return m_external_ip.external_address(address_v4());
+#if TORRENT_USE_IPV6
```

Comment:
this interleaving is hard to read I think. could you initialize a variable address, to either v4 or v6 and then do the call unconditionally? and maybe coalesce the #if/#endif blocks